presto-main/src/main/java/com/facebook/presto/sql/planner/NodePartitioningManager.java

```diff
@@ -226,4 +224,30 @@ private ConnectorBucketNodeMap getConnectorBucketNodeMap(Session session, Partit
         }
         return distribution.build();
     }
+
+    private static List<InternalNode> createAffinityBucketToNode(List<InternalNode> nodes, int bucketCount)
+    {
+        List<InternalNode> sortedNodes = nodes.stream()
+                .sorted(comparing(InternalNode::getNodeIdentifier))
+                .collect(toImmutableList());
+        return Stream.generate(() -> sortedNodes).flatMap(List::stream).limit(bucketCount).collect(toImmutableList());
+    }
+
+    private List<InternalNode> calculateBucketToNode(ConnectorBucketNodeMap connectorBucketNodeMap, PartitioningHandle partitioningHandle, Session session)
+    {
+        ConnectorId connectorId = partitioningHandle.getConnectorId()
+                .orElseThrow(() -> new IllegalArgumentException("No connector ID for partitioning handle: " + partitioningHandle));
+
+        ConnectorNodePartitioningProvider partitioningProvider = partitioningProviderManager.getPartitioningProvider(connectorId);
+        Set<NodeSelectionStrategy> supportedNodeSelectionStrategies = partitioningProvider.listSupportedNodeSelectionStrategies(session.toConnectorSession(connectorId));
+        Optional<NodeSelectionStrategy> nodeSelectionStrategy = nodeScheduler.getNodeSelectionStrategy();
+        if (nodeSelectionStrategy.isPresent()
+                && nodeSelectionStrategy.get() == NodeSelectionStrategy.AFFINITY
+                && supportedNodeSelectionStrategies.contains(nodeSelectionStrategy.get())) {
+            return createAffinityBucketToNode(nodeScheduler.createNodeSelector(connectorId).selectRandomNodes(getMaxTasksPerStage(session)), connectorBucketNodeMap.getBucketCount());
```

Comment:
I was originally thinking how to deal with this; but this was already an issue for existing connectors with non-`isRemoteAccessible` like Raptor. So for `SOFT_`/`HARD_AFFINITY` case, we may use the old `allNodes()`.
cc: @wenleix @arhimondr @shixuan-fan

Reply:
I'm a bit confused. What do you mean by "this" in "deal with this"?

Reply:
I mean max task per stage for nodes picking.